### DISCUSSION TITLE
I want to know, how can I close the program in python?

### DISCUSSION THREAD

**Discussion by iceriny** [r0x0r/pywebview#1415] [Q&A] [answered]
**Help!**
I want to know, how can I close the program in python? - And be able to trigger the `closed|closing event`.
I use `webview.windows[0].destroy()` Even if there are no other windows, the program background will not end.
```python
class Api:

    def close_window(self) -> None:
        Debug("close_window", webview.windows[0].title)
        webview.windows[0].destroy()
```

**Comment by xandaaah** [comment]
```
import os
os._exit(0)
```
This should close the program and any windows created by it.

**Comment by r0x0r** [comment, marked as answer]
@iceriny when you destroy the last window it will release webview.start and kill the gui loop. If you have nothing else blocking the execution like non-daemon threads your program will quit.
